warehouse/query-core/src/test/java/datawave/query/index/lookup/TupleToRangeTest.java

```diff
@@ -1,87 +1,235 @@
+import java.util.List;
+import java.util.Set;
+
+import static datawave.common.test.utils.query.RangeFactoryForTests.makeDayRange;
+import static datawave.common.test.utils.query.RangeFactoryForTests.makeShardedRange;
+import static datawave.common.test.utils.query.RangeFactoryForTests.makeTestRange;
+import static datawave.common.test.utils.query.RangeFactoryForTests.makeTldTestRange;
 
-import static datawave.common.test.utils.query.RangeFactory.*;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertFalse;
 import static org.junit.Assert.assertTrue;
 
 public class TupleToRangeTest {
     
-    protected JexlNode queryNode = null;
+    private JexlNode queryNode = null;
+    private ShardQueryConfiguration config;
     
     @Before
     public void setup() throws ParseException {
         queryNode = JexlASTHelper.parseJexlQuery("true==true");
+        config = new ShardQueryConfiguration();
     }
     
     @Test
-    public void test() {
-        IndexInfo info = new IndexInfo(Lists.newArrayList(new IndexMatch("a"), new IndexMatch("b"), new IndexMatch("c"), new IndexMatch("d")));
-        info.applyNode(queryNode);
-        Tuple2<String,IndexInfo> tuple = Tuples.tuple("s", info);
-        Iterator<QueryPlan> ranges = new TupleToRange(queryNode, new ShardQueryConfiguration()).apply(tuple);
-        assertTrue(ranges.hasNext());
-        assertEquals(makeTestRange("s", "a"), ranges.next().getRanges().iterator().next());
-        assertTrue(ranges.hasNext());
-        assertEquals(makeTestRange("s", "b"), ranges.next().getRanges().iterator().next());
-        assertTrue(ranges.hasNext());
-        assertEquals(makeTestRange("s", "c"), ranges.next().getRanges().iterator().next());
-        assertTrue(ranges.hasNext());
-        assertEquals(makeTestRange("s", "d"), ranges.next().getRanges().iterator().next());
-        assertFalse(ranges.hasNext());
+    public void testIsDocumentRange() {
+        Set<String> docIds = Sets.newHashSet("docId0", "docId1", "docId2");
+        IndexInfo indexInfo = new IndexInfo(docIds);
+        
+        TupleToRange tupleToRange = new TupleToRange(null, null);
+        assertTrue(tupleToRange.isDocumentRange(indexInfo));
+        
+        IndexInfo otherInfo = new IndexInfo(3L);
+        assertFalse(tupleToRange.isDocumentRange(otherInfo));
     }
     
     @Test
-    public void testTld() {
-        IndexInfo info = new IndexInfo(Lists.newArrayList(new IndexMatch("a"), new IndexMatch("b"), new IndexMatch("c"), new IndexMatch("d")));
-        info.applyNode(queryNode);
-        Tuple2<String,IndexInfo> tuple = Tuples.tuple("s", info);
-        ShardQueryConfiguration config = new ShardQueryConfiguration();
+    public void testIsShardRange() {
+        String shardRange = "20190314_0";
+        String dayRange = "20190314";
+        
+        TupleToRange tupleToRange = new TupleToRange(null, null);
+        assertTrue(tupleToRange.isShardRange(shardRange));
```

Comment:
It reads a little strange that the isShardRange, isDocumentRange, etc. are TupleToRange methods. The implementations of those methods don't use any state from the TupleToRange instance. What if we made the test methods static methods on your RangeFactory? Like `RangeFactory.isShardRange(shardRange)`.

Reply:
The tricky part here is these methods are specific implementations of the TupleToRange's function of transforming index information to ranges for query. The RangeFactory's sole responsibility is to build ranges, I think adding identity method like isShardRange could clutter the class.

Put another way, the TupleToRange is responsible for identifying and extracting information to build the ranges, the RangeFactory ensures those ranges are built in a standard way.

Reply:
The building of a range and the testing of a range are logically tied together concepts. Consider if we changed what a shard range looks like. You'd have to modify the creation of it in RangeFactory and the testing of it in TupleToRange. I think the fact that you need to instantiate a `new TupleToRange(null, null)` to call this method is another hint that it may be in the wrong place. You could have called `new TupleToRange(someNode, someConfig)` and it'd have no impact on the result of the `isShardRange` and `isDocumentRange` function call. My first preference would be to move them to RangeFactory. If you believe they should be in TupleToRange, I would ask we at least make them static.

Reply:
Let's go with the static option